### PR TITLE
Entities upgrade 0.3.0

### PR DESCRIPTION
**Pull request by zeroZshadow**
#### Description
Upgrade the ECS entities packages to 0.3.0

#### Tests
- [x] Playground
- [x] Test Project
- [x] Standalone

#### Documentation
- [x] Changelog
- [ ] Upgrade guide deluxe

### CODE REVIEW

**Comment by zeroZshadow** [comment]
Step 1: Upgraded to 0.2.0-preview.18 so far

File: workers/unity/Packages/io.improbable.gdk.transformsynchronization/TransformUtils.cs [hunk 1]
```diff
@@ -66,22 +66,35 @@ public static CompressedQuaternion ToCompressedQuaternion(this Quaternion quater
...
 
+            if (!requireAuthority)
+            {
+                componentQueryDesc.None = new[]
+                {
+                    ComponentType.ReadOnly<TransformInternal.Authoritative>()
+                };
+            }
+
             return componentQueryDesc;
```
Comment (jamiebrynes7) [inline workers/unity/Packages/io.improbable.gdk.transformsynchronization/TransformUtils.cs]: How useful is this function now? Like effectively all it does is add `T` to all and then the `TransformInternal.Authoritative` to either the `All` or `None` depending on the bool (which is 😕 already)

File: workers/unity/Packages/io.improbable.gdk.transformsynchronization/Systems/DefaultApplyLatestTransformSystem.cs [hunk 2]
```diff
@@ -44,13 +42,15 @@ internal void RegisterTransformSyncType<T>(ITransformSync<T> impl)
 
         private void UpdateTransformQuery()
         {
-            var transformQueryDesc = TransformUtils.ConstructEntityQueryDesc<UnityEngine.Transform>(baseComponentTypes);
-            transformQueryDesc.None = applyLatestTransformActions.Keys
-                .Select(ComponentType.ReadOnly)
+            var transformQueryDesc = TransformUtils.ConstructEntityQueryDesc<UnityEngine.Transform>(requireAuthority: false, baseComponentTypes);
+            transformQueryDesc.None = transformQueryDesc.None
+                .Union(
```
[the remark this answers is not in the document]
Comment (zeroZshadow) [reply workers/unity/Packages/io.improbable.gdk.transformsynchronization/Systems/DefaultApplyLatestTransformSystem.cs]: It only needs a union if the `ConstructEntityQueryDesc` does NOT require auth.
Its weird, I know, the transform sync needs to have an overhaul.

File: workers/unity/Packages/io.improbable.gdk.transformsynchronization/TransformUtils.cs (hunk 1 above)
Comment (zeroZshadow) [reply workers/unity/Packages/io.improbable.gdk.transformsynchronization/TransformUtils.cs]: I'm considering replacing this whole thing, but I think thats for later.

**Comment by zeroZshadow** [comment]
Upgraded to 0.3.0 instead, required no changes and fixed a crash in standalone builds

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,11 @@
 ### Breaking Changes
 
 - The minimum version of Unity supported by the GDK is now 2019.3. You will need to upgrade your Unity project. [#1336](https://github.com/spatialos/gdk-for-unity/pull/1336)
+- Upgrade the com.unity.entities packages to 0.3.0-preview.4. [#1337](https://github.com/spatialos/gdk-for-unity/pull/1337)
+    - Authority in ECS is no longer a SharedComponentData, but instead a tag component.
+    - `ComponentAuthority` has been renamed to `HasAuthority`.
+    - No need to filter your ECS query anymore, instead match for `HasAuthority`.
+    - build-in Unity AssetBundle module is now required.
```
Comment (paulbalaji) [inline CHANGELOG.md]:
```suggestion
- Upgraded the `com.unity.entities` packages to `0.3.0-preview.4`. [#1337](https://github.com/spatialos/gdk-for-unity/pull/1337)
    - Authority in ECS is no longer a `SharedComponentData`, but instead a tag component.
    - `ComponentAuthority` has been renamed to `HasAuthority`.
    - No need to filter your ECS query anymore, instead match for `HasAuthority`.
    - Built-in Unity AssetBundle module is now required.
```

File: workers/unity/Packages/io.improbable.gdk.transformsynchronization/Systems/DefaultApplyLatestTransformSystem.cs [hunk 4]
```diff
@@ -44,13 +42,15 @@ internal void RegisterTransformSyncType<T>(ITransformSync<T> impl)
...
+                    applyLatestTransformActions.Keys
+                        .Select(ComponentType.ReadOnly)
+                        .ToArray())
```
[the remark this answers is not in the document]
Comment (zeroZshadow) [reply workers/unity/Packages/io.improbable.gdk.transformsynchronization/Systems/DefaultApplyLatestTransformSystem.cs]: Will do

**Pull request merged**